test_conformance/extensions/cl_khr_command_buffer/negative_tests/command_buffer_create.cpp

```diff
@@ -0,0 +1,318 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "../basic_command_buffer.h"
+#include "procs.h"
+
+
+//--------------------------------------------------------------------------
+namespace {
+
+// CL_INVALID_VALUE if num_queues is not one.
+struct CreateCommandBufferNumQueues : public BasicCommandBufferTest
+{
+    // using BasicCommandBufferTest::BasicCommandBufferTest;
+    CreateCommandBufferNumQueues(cl_device_id device, cl_context context,
+                                 cl_command_queue queue)
+        : BasicCommandBufferTest(device, context, queue), queue1(nullptr),
+          queue2(nullptr)
+    {}
+
+    cl_int Run() override
+    {
+        cl_int error = CL_SUCCESS;
+
+        cl_command_queue queues[2] = { queue1, queue2 };
+
+        command_buffer = clCreateCommandBufferKHR(2, queues, nullptr, &error);
+
+        test_failure_error_ret(
+            error, CL_INVALID_VALUE,
+            "clCreateCommandBufferKHR should return CL_INVALID_VALUE",
+            TEST_FAIL);
+
+        return CL_SUCCESS;
+    }
+
+    cl_int SetUp(int elements) override
+    {
+        cl_int error = CL_SUCCESS;
+
+        error = BasicCommandBufferTest::SetUp(elements);
+        test_error(error, "BasicCommandBufferTest::SetUp failed");
+
+        queue1 = clCreateCommandQueue(context, device, 0, &error);
+        test_error(error, "clCreateCommandQueue failed");
+        queue2 = clCreateCommandQueue(context, device, 0, &error);
+        test_error(error, "clCreateCommandQueue failed");
+
+        return CL_SUCCESS;
+    }
+
+    clCommandQueueWrapper queue1;
+    clCommandQueueWrapper queue2;
+};
+
+// CL_INVALID_VALUE if queues is NULL.
+struct CreateCommandBufferNullQueues : public BasicCommandBufferTest
+{
+    using BasicCommandBufferTest::BasicCommandBufferTest;
+
+    cl_int Run() override
+    {
+        cl_int error = CL_SUCCESS;
+
+        command_buffer = clCreateCommandBufferKHR(1, nullptr, nullptr, &error);
+
+        test_failure_error_ret(
+            error, CL_INVALID_VALUE,
+            "clCreateCommandBufferKHR should return CL_INVALID_VALUE",
+            TEST_FAIL);
+
+        return CL_SUCCESS;
+    }
+};
+
+// CL_INVALID_VALUE if values specified in properties are not valid,
+// or if the same property name is specified more than once.
+struct CreateCommandBufferRepeatedProperties : public BasicCommandBufferTest
+{
+    using BasicCommandBufferTest::BasicCommandBufferTest;
+
+    cl_int Run() override
+    {
+        cl_int error = CL_SUCCESS;
+
+        cl_command_buffer_properties_khr repeated_properties[4] = {
+            CL_COMMAND_BUFFER_FLAGS_KHR, CL_COMMAND_BUFFER_SIMULTANEOUS_USE_KHR,
+            CL_COMMAND_BUFFER_SIMULTANEOUS_USE_KHR, 0
```

Comment:
I think the property *name* here is `CL_COMMAND_BUFFER_FLAGS_KHR`. So this is testing an invalid property name which is good, but think also testing for a repeated but valid property could be done. e.g.
```
cl_command_buffer_properties_khr repeated_properties[5] = {
 CL_COMMAND_BUFFER_FLAGS_KHR, CL_COMMAND_BUFFER_SIMULTANEOUS_USE_KHR,
 CL_COMMAND_BUFFER_FLAGS_KHR, CL_COMMAND_BUFFER_SIMULTANEOUS_USE_KHR, 0};
```